Add TOTAL_MEMORY_ALLOCATED profiling metric

krleonid opened the pull request:
This PR adds a new profiling metric `TOTAL_MEMORY_ALLOCATED` that tracks the total amount of memory allocated during query execution.

## Changes

- Added `TOTAL_MEMORY_ALLOCATED` to `MetricsType` enum and `query_global_metrics`
- Implemented memory tracking through `ClientBufferManager` wrapper
- Updated `QueryProfiler` to collect and report `total_memory_allocated`
- Included metric in default profiling settings
- Added comprehensive test coverage

## Implementation Details

The implementation tracks memory allocations through the `ClientBufferManager` wrapper, which intercepts all memory allocation calls and forwards them to `QueryProfiler::AddToCounter`. The metric is treated as a query-global metric, meaning it only appears at the root level of the profiling tree, not in individual operator nodes.

## Testing

- Added new test file: `test_custom_profiling_total_memory_allocated.test`
- All existing profiling tests pass

krleonid committed:
Add TOTAL_MEMORY_ALLOCATED profiling metric

- Add TOTAL_MEMORY_ALLOCATED to MetricsType enum and query_global_metrics
- Track memory allocations through ClientBufferManager wrapper
- Update QueryProfiler to collect and report total_memory_allocated
- Include metric in default profiling settings
- Add comprehensive test coverage for the new metric
- Ensure metric only appears at root level (not in children nodes)

Based on PR #17699

krleonid committed:
Fix MetricsType enum count from 64 to 65

Update the enum count to reflect the addition of TOTAL_MEMORY_ALLOCATED

maiadegraaf commented on test/sql/pragma/profiling/test_custom_profiling_total_memory_allocated.test:
```diff
@@ -0,0 +1,152 @@
+# Verify that children nodes don't have total_memory_allocated
+# We'll check by looking at the JSON structure - children should not have this field
+statement ok
+CREATE OR REPLACE TABLE json_data AS SELECT * FROM '__TEST_DIR__/profiling_output.json';
```
I don't think this test checks if children have `total_memory_allocated`.

```sql
CREATE OR REPLACE TABLE metrics_children AS SELECT unnest(children, recursive := true) FROM metrics_output;
SELECT total_memory_allocated FROM metrics_children;
```

maiadegraaf reviewed (commented):
Thanks for the PR! Great expansive testing. Looks good to me overall. Just added a comment.

krleonid committed:
Add verification that children nodes don't have total_memory_allocated

Use unnest with recursive to flatten all children nodes and verify
that total_memory_allocated field does not exist in any children nodes.

krleonid committed:
Remove redundant comment from test

maiadegraaf reviewed (approved):
Thanks for the fix! Looks good from my side.
